svd-rs/src/array.rs

```diff
@@ -50,11 +50,51 @@ where
     }
 }
 
+impl<T> Description for MaybeArray<T>
+where
+    T: Description,
+{
+    fn description(&self) -> Option<&str> {
+        T::description(self)
+    }
+}
+
 /// Return list of names of instances in array
 pub fn names<'a, T: Name>(info: &'a T, dim: &'a DimElement) -> impl Iterator<Item = String> + 'a {
     let name = info.name();
-    dim.indexes()
-        .map(move |i| name.replace("[%s]", &i).replace("%s", &i))
+    dim.indexes().map(move |i| {
+        dim.dim_array_index
+            .as_ref()
+            .map(|dai| {
+                dai.values
+                    .iter()
+                    .find(|e| e.value.map(|v| v.to_string().as_str() == i.deref()) == Some(true))
+            })
+            .flatten()
+            .map(|n| n.name.clone())
+            .unwrap_or_else(|| name.replace("[%s]", &i).replace("%s", &i))
+    })
+}
+
+/// Return list of descriptions of instances in array
+pub fn descriptions<'a, T: Description>(
+    info: &'a T,
+    dim: &'a DimElement,
+) -> impl Iterator<Item = Option<String>> + 'a {
+    let description = info.description();
+    dim.indexes().map(move |i| {
+        dim.dim_array_index
+            .as_ref()
+            .map(|dai| {
+                dai.values
+                    .iter()
+                    .find(|e| e.value.map(|v| v.to_string().as_str() == i.deref()) == Some(true))
+            })
+            .flatten()
+            .map(|n| n.description.clone())
+            .flatten()
```

Comment:
`Some(_).map(...).flatten()` is the same as `and_then`

Reply:
Exactly. Thanks.